### PR TITLE
Install new python dependencies during daily maintenance

### PR DESCRIPTION
**Pull request by deajan**
As of today, check_requirements simply checks for `PyMySQL` package, whereas `requirements.txt` lists a couple more dependencies.
This makes the checks dynamically read `requirements.txt`

WARNING: as of today, this would make `daily.sh` complain with something like the following on elder 1.5.x librenms setups:
```
[FAIL]  Python3 module issue found: '('Required packages:', ['PyMySQL', 'python-dotenv', 'redis>=3.0', 'setuptools', 'psutil'])
Package not found: psutil
```
I think that just like the  `composer install` in `daily.sh`, there should be a `pip3 install -r requirements` in `daily.sh` in order to make sure we update packages as we update LibreNMS.

I would be happy to modify the script if the LibreNMS dev team is okay with the above statement.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [X] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Review by murrant — CHANGES_REQUESTED** [review]
If you check the history of this file, it originally did this.

It was changed to just check PyMySQL because this is used to put users on the python2 support branch... 

They dynamic dependencies had a lot of false positives pushing users to the python2 support branch when they had no need of redis for example.

If you want to check this dynamically, it needs to be in a different file.  Also, LibreNMS no longer supports Python 2, so you don't need to add compatibility options.

**Comment by deajan** [comment]
Thanks for the clarification.
How about we just have a static list when running python2 and read `requirements.txt` for python3 ?
Would that fit ?

**Comment by murrant** [comment]
basically what you need to do is revert check_requirements.py and put your script in a new file.

File: daily.sh [hunk 1]
```diff
@@ -136,8 +136,8 @@ check_dependencies() {
 
     python3=$(python3 -c "import sys;print(int(sys.version_info < (3, 4)))" 2> /dev/null)
     if [ "$python3" -eq 0 ]; then
-        python3 scripts/check_requirements.py > /dev/null 2>&1 || $(type -p python3) -m pip install -r requirements.txt > /dev/null 2>&1
```
Comment (murrant) [inline daily.sh]: Do not change daily.sh  I'm telling you it will break installs.
Only add your requirements check in validation.

File: daily.sh [hunk 2]
```diff
@@ -133,14 +133,27 @@ check_dependencies() {
     local branch ver_56 ver_71 ver_72 ver_73 python3 python_deps phpver pythonver old_branches msg
 
     branch=$(git rev-parse --abbrev-ref HEAD)
-    scripts/check_requirements.py > /dev/null 2>&1 || pip3 install -r requirements.txt > /dev/null 2>&1
+
+    python3=$(python3 -c "import sys;print(int(sys.version_info < (3, 4)))" 2> /dev/null)
```
Comment (murrant) [inline daily.sh:137]: Don't check this in daily.sh, only in validation.

Comment (deajan) [reply daily.sh:137]: I am sorry, I don't get it. In th daily.sh version on master, we already have check_requirepents.py || pip install -r requirements.py
I just split into python2 mode that stays the same, and python3 that adds more checks.
Running pip as user does not require privileges and will install necessary packages.

What does it break exactly ? Sorry if this doesn't make sense to you :(

**Comment by Jellyfrog** [comment]
My two cents, remove the downgrade functionally in daily.sh for python deps, keep only python version check.
We're not downgrading if composer fails, so why do it if python deps fail?

Other related ramblings;
daily.sh is a mess, and its hard to know what is a "validation" vs an "installation".
Basically we should try to do as little "possible breaking stuff" as possible in daily.sh

Here is how I see it in the future:
Laravel scheduler takes care of all maintenance tasks like database cleanup etc, basically all this stuff:
```
                status_run 'Updating SQL-Schema' 'php includes/sql-schema/update.php'
                status_run 'Updating submodules' "$DAILY_SCRIPT submodules"
                status_run 'Cleaning up DB' "$DAILY_SCRIPT cleanup"
                status_run 'Fetching notifications' "$DAILY_SCRIPT notifications"
                status_run 'Caching PeeringDB data' "$DAILY_SCRIPT peeringdb"
                status_run 'Caching Mac OUI data' "$DAILY_SCRIPT mac_oui"
```

A separate update.sh script should be created, only dealing with upgrades, in a more robust way.

1. Fetch changes without merging (ie. git fetch)
2. Check if we would fulfil the requirements of the new code (php, python, net-snmp, ... )
3. Switch to the code
4. Upgrade dependencies
5. Run migrations

**Comment by murrant** [comment]
We added the dependency check because a lot of people were missing the python3 mysql package required for polling.  I'm fine with hardcoding that specific check (I don't think we can remove it yet, people still run crazy old versions)

**Comment by murrant** [comment]
@deajan fixing this up because it is needed :)

Don't fail during, daily.sh, install python deps at the normal spot during composer install.  Use for validation too.

**Comment by murrant** [comment]
originally, it was a dynamic check, but then it needed to only check pymsql for daily.sh.  Splitting them out updating the places that expected a dynamic check is the correct fix.

**Review by murrant — APPROVED** [review]

**Comment by deajan** [comment]
Sorry for not being very active, currently moving my housing... no time for computers these weeks ;(

**Comment by murrant** [comment]
No worries @deajan only a small change to merge this.

**Pull request merged**